external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -148,18 +149,28 @@ private[kinesis] class KinesisReceiver[T](
 
     kinesisCheckpointer = new KinesisCheckpointer(receiver, checkpointInterval, workerId)
     val kinesisProvider = kinesisCreds.provider
-    val kinesisClientLibConfiguration = new KinesisClientLibConfiguration(
+    var kinesisClientLibConfiguration = new KinesisClientLibConfiguration(
```

Comment:
Keep this a val, but you can introduce a new scope with a temp val using braces, e.g.: 

```scala
val kinesisClientLibConfiguration = {
  val baseClientLibConfiguration = new KinesisClientLibConfiguration(
      checkpointAppName,
      streamName,
      ...
    .withKinesisEndpoint(endpointUrl)
    .withInitialPositionInStream(initialPosition.initialPositionInStream)
    ...

  initialPosition match {
    // see comment below
    ...
  }
}
```